bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -139,6 +160,39 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
         }
     }
 
+    private Collection<Resource> getAssetFolders(Page page, ResourceResolver resolver) {
+        Map<String, Resource> allAssetFolders = new TreeMap<String, Resource>();
+        ValueMap properties = page.getProperties();
+        String[] configuredAssetFolderPaths = properties.get(damAssetProperty, String[].class);
+        if (configuredAssetFolderPaths != null) {
+            for (String configuredAssetFolderPath : configuredAssetFolderPaths) {
+                if (StringUtils.isNotBlank(configuredAssetFolderPath)) {
+                    Resource configuredAssetFolder = resolver.getResource(configuredAssetFolderPath);
+                    if (configuredAssetFolder != null && configuredAssetFolder.isResourceType("sling:OrderedFolder")) {
+                        for (Resource assetFolder : getAssetFolders(configuredAssetFolder)) {
+                            if (!allAssetFolders.containsKey(assetFolder.getPath())) {
+                                allAssetFolders.put(assetFolder.getPath(), assetFolder);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return allAssetFolders.values();
+    }
+
+    private List<Resource> getAssetFolders(Resource assetFolder) {
+        List<Resource> damFolders = new ArrayList<Resource>();
+        damFolders.add(assetFolder);
+        for (Iterator<Resource> children = assetFolder.listChildren(); children.hasNext();) {
+            Resource assetFolderChild = children.next();
+            if (assetFolderChild.isResourceType("sling:OrderedFolder")) {
```

Comment:
I see the comments above now too, regarding simply checking all children if they are DAM Assets.  I suppose that would work...just didn't know what constructs other than folders would contain Assets AND authors would expect those items to then be included in the sitemap.

One thing worth noting ... I collected asset folders first so that I could sift out duplicates.  I could do it all in one shot (filtering out duplicate assets as we go) this just felt a little more clean in that here we gather all the folders and handle dupes, then the other code simply handles just writing the assets (similar to how code is "just writing" the pages) to the sitemap.